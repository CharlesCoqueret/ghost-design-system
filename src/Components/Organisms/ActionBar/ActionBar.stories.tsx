import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import ActionBar, { IActionBarProps } from './ActionBar';
import { ColorButtonEnum } from '../../Molecules';

export default {
  title: 'Organism/ActionBar',
  component: ActionBar,
} as ComponentMeta<typeof ActionBar>;

const Template: ComponentStory<typeof ActionBar> = (args: IActionBarProps) => {
  return <ActionBar {...args} />;
};

export const Default = Template.bind({});
Default.args = {
  prefix: 'prefix',
  title: 'title',
  suffix: 'suffix',
  placeholder: 'placeholder',
  entityId: 'entityId',
  onTitleEdit: (newTitle) => {
    console.log(newTitle);
  },
  actions: [
    {
      label: 'Cancel',
      color: ColorButtonEnum.SECONDARY,
      onClick: () => {
        console.log('Cancel clicked');
      },
    },
    {
      label: 'Submit',
      icon: ['fal', 'paper-plane'],
      color: ColorButtonEnum.PRIMARY,
      onClick: () => {
        console.log('Submit clicked');
      },
    },
  ],
  basicActions: [
    {
      tooltip: 'Save',
      color: ColorButtonEnum.REVERSED,
      icon: ['fal', 'save'],
      onClick: () => {
        console.log('Save clicked');
      },
    },
    {
      tooltip: 'Undo',
      icon: ['fal', 'undo'],
      color: ColorButtonEnum.REVERSED,
      onClick: () => {
        console.log('Undo clicked');
      },
    },
  ],
};
