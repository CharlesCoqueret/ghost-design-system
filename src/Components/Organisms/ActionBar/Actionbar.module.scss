.container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding-left: 20px;
  padding-right: 20px;
  height: 50px;
}

.primary {
  position: sticky;
  box-sizing: border-box;
  box-shadow: 0px 4px 10px rgba(var(--theme-provider-black), 0.1);
  height: 64px;
  margin-bottom: 10px;
}

.secondary {
  position: sticky;
  &:before {
    border-bottom: 1px solid rgb(var(--theme-provider-chalk));
    bottom: 0;
    content: '';
    height: 1px;
    left: 20px;
    position: absolute;
    right: 20px;
  }
}

.left {
  display: flex;
  flex-direction: row;
  overflow: hidden;
}

.back {
  color: rgb(var(--theme-provider-charcoal));
  margin: auto 5px auto 0px;
}

.badges {
  display: flex;
  color: rgb(var(--theme-provider-charcoal));
  padding: auto 5px;
  & > * {
    margin: auto 5px;
  }
}

.right {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.actions {
  display: flex;
  & > * {
    margin: auto 5px;
  }
}

.separator {
  display: flex;
  margin: auto 10px;
}

.basicActions {
  display: flex;
  & > * {
    margin: auto 5px;
  }
}
