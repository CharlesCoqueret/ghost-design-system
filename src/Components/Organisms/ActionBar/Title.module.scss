.container {
  display: inline-flex;
  overflow: hidden;
}

.prefix,
.entity,
.suffix {
  border: 1px solid transparent;
  margin-left: 2px;
  margin-right: 2px;
}

.hiddenTitle {
  background-color: transparent;
  border-radius: 4px;
  border: 1px solid transparent;
  color: transparent;
  display: inline-flex;
  font-size: 22px;
  margin: auto 2px;
  outline: none;
  overflow: hidden;
  position: absolute;
  text-overflow: ellipsis;
  text-transform: uppercase;
  user-select: none;
  white-space: nowrap;
  z-index: -1;
}

.title {
  background-color: transparent;
  border-radius: 4px;
  border: 1px solid transparent;
  display: inline-flex;
  font-size: 22px;
  margin: auto 2px;
  outline: none;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;

  &:not(:read-only):hover {
    border: 1px solid rgb(var(--theme-provider-chalk));
  }

  &:not(:read-only):focus {
    border: 1px solid rgb(var(--theme-provider-scooter));
  }
}
