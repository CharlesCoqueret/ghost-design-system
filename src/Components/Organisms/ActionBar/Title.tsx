import React, {
  ChangeEvent,
  CSSProperties,
  FocusEvent,
  KeyboardEvent,
  ReactElement,
  useCallback,
  useEffect,
  useRef,
  useState,
} from 'react';

import { Tooltip } from '../../Atoms/Tooltip';
import { Typography } from '../../Atoms/Typography';

export interface ITitleProps {
  dataTestId?: string;
  entityId?: string;
  onTitleEdit?: (newTitle: string) => void;
  placeholder?: string;
  prefix?: string;
  renameTooltip?: string;
  suffix?: string;
  title?: string;
}

const Title = (props: ITitleProps): ReactElement => {
  const { dataTestId, entityId, onTitleEdit, placeholder, prefix, renameTooltip, suffix, title } = props;

  const [currentTitle, setCurrentTitle] = useState<string | undefined>(title);
  const [currentInputStyle, setCurrentInputStyle] = useState<CSSProperties>({ display: 'flex', width: '0px' });
  const spanRef = useRef<HTMLSpanElement>(null);
  const inputRef = useRef<HTMLInputElement>(null);

  const isEditable = onTitleEdit !== undefined;

  const updateInputWidth = useCallback(() => {
    setCurrentInputStyle({
      display: 'flex',
      margin: 'auto',
      width: `${spanRef.current?.offsetWidth ? spanRef.current.offsetWidth + 1 : 0}px`,
    });
  }, [spanRef]);

  const handleKeyDown = useCallback((event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter' || event.key === 'Tab') {
      event.currentTarget.blur();
    }
  }, []);

  const handleChange = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => {
      setCurrentTitle(event.target.value);
      updateInputWidth();
    },
    [spanRef],
  );

  const handleBlur = useCallback(
    (event: FocusEvent<HTMLInputElement>) => {
      if (event.target.value.trim().length == 0) {
        setCurrentTitle(title);
        return;
      }
      if (onTitleEdit) {
        onTitleEdit(event.target.value.trim());
      }
    },
    [currentTitle],
  );

  useEffect(() => {
    updateInputWidth();
  }, [currentTitle]);

  return (
    <>
      {prefix && (
        <Typography.Title level={1} ellipsis className='align-edit'>
          {prefix}
        </Typography.Title>
      )}
      <span ref={spanRef} className={'title-edit-hidden'}>
        {currentTitle || placeholder}
      </span>
      <Tooltip tooltip={isEditable ? renameTooltip : undefined}>
        <input
          autoComplete='off'
          autoFocus={false}
          className='title-edit'
          data-testid={dataTestId}
          onBlur={handleBlur}
          onChange={handleChange}
          onKeyDown={handleKeyDown}
          placeholder={placeholder}
          readOnly={!isEditable}
          ref={inputRef}
          style={currentInputStyle}
          type='text'
          tabIndex={isEditable ? 0 : -1}
          value={currentTitle}
        />
      </Tooltip>
      {entityId && (
        <Typography.Title level={1} ellipsis className='align-edit'>
          - {entityId}
        </Typography.Title>
      )}
      {suffix && (
        <Typography.Title level={1} ellipsis className='align-edit'>
          {suffix}
        </Typography.Title>
      )}
    </>
  );
};

Title.defaultProps = {
  entityId: undefined,
  onTitleEdit: undefined,
  placeholder: undefined,
  prefix: undefined,
  renameTooltip: 'Rename',
  suffix: undefined,
  title: '',
};

export default Title;
