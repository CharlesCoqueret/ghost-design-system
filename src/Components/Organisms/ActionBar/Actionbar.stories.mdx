import { Canvas, Meta, ArgsTable, Story, Source } from '@storybook/addon-docs';

import ActionBar from './ActionBar';
import { ColorButtonEnum } from '../../Molecules';
import { Badge, BadgeColorsEnum, Icon, Tooltip } from '../../Atoms';

export const Template = (args) => {
  return <ActionBar {...args} />;
};

<Meta
  title='Organism'
  component={ActionBar}
  parameters={{
    viewMode: 'docs',
    layout: 'fullscreen',
  }}
/>

# ActionBar

The action bar provides to the user context and actions related to the current page scope.

Play with ActionBar live on [CodeSandbox](https://codesandbox.io/s/actionbar-demo-k8y5tr)

For layout management, please refer to this [example](https://codesandbox.io/s/size-management-sidebar-xtpkk0)

<Canvas withSource='open'>
  <Story
    name='ActionBar'
    args={{
      prefix: 'prefix',
      title: 'title',
      suffix: 'suffix',
      primary: true,
      placeholder: 'placeholder',
      entityId: 'entityId',
      onBackClicked: () => {
        console.log('Back clicked');
      },
      backTooltip: 'Back',
      onTitleEdit: (newTitle) => {
        console.log(newTitle);
      },
      icon: <Icon icon={['fal', 'question-circle']} size='lg' />,
      indicator: (
        <Badge color={BadgeColorsEnum.HIGH} tooltip='indicator' type='notification'>
          1
        </Badge>
      ),
      status: (
        <Badge color={BadgeColorsEnum.SECONDARY} tooltip='status' type='indicator'>
          Status
        </Badge>
      ),
      actions: [
        {
          label: 'Submit',
          icon: ['fal', 'paper-plane'],
          color: ColorButtonEnum.PRIMARY,
          onClick: () => {
            console.log('Submit clicked');
          },
        },
        {
          label: 'Cancel',
          color: ColorButtonEnum.SECONDARY,
          onClick: () => {
            console.log('Cancel clicked');
          },
        },
      ],
      basicActions: [
        {
          tooltip: 'Save',
          color: ColorButtonEnum.REVERSED,
          icon: ['fal', 'save'],
          onClick: () => {
            console.log('Save clicked');
          },
        },
      ],
    }}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable of={ActionBar} sort='requiredFirst' />

### Title

<Canvas withSource='open'>
  <ActionBar title='title' />
</Canvas>

### Primary

<Canvas withSource='open'>
  <div>
    <ActionBar title='Primary' primary={true} />
    <ActionBar title='Secondary' primary={false} />
  </div>
</Canvas>

### Prefix, suffix and entityId

<Canvas withSource='open'>
  <ActionBar title='title' entityId='entityId' prefix='prefix' suffix='suffix' />
</Canvas>

### Back

<Canvas withSource='open'>
  <ActionBar
    title='title'
    onBackClicked={() => {
      console.log('Back clicked');
    }}
    backTooltip='Back'
  />
</Canvas>

### Title edition

<Canvas withSource='open'>
  <ActionBar
    title='title'
    onTitleEdit={(newTitle) => {
      console.log(newTitle);
    }}
    placeholder='placeholder'
  />
</Canvas>

### Icon

<Canvas withSource='open'>
  <ActionBar title='title' icon={<Icon icon={['fal', 'question-circle']} size='lg' />} />
</Canvas>

### Indicator

<Canvas withSource='open'>
  <ActionBar
    title='title'
    indicator={
      <Badge color={BadgeColorsEnum.HIGH} tooltip='indicator' type='notification'>
        1
      </Badge>
    }
  />
</Canvas>

### Status

<Canvas withSource='open'>
  <ActionBar
    title='title'
    status={
      <Badge color={BadgeColorsEnum.SECONDARY} tooltip='status' type='indicator'>
        Status
      </Badge>
    }
  />
</Canvas>

### Full size action

<Canvas withSource='open'>
  <ActionBar
    title='title'
    actions={[
      {
        label: 'Submit',
        icon: ['fal', 'paper-plane'],
        color: ColorButtonEnum.PRIMARY,
        onClick: () => {
          console.log('Submit clicked');
        },
      },
      {
        label: 'Cancel',
        color: ColorButtonEnum.SECONDARY,
        onClick: () => {
          console.log('Cancel clicked');
        },
      },
    ]}
  />
</Canvas>

### Short hand actions

<Canvas withSource='open'>
  <ActionBar
    title='title'
    basicActions={[
      {
        tooltip: 'Save',
        color: ColorButtonEnum.REVERSED,
        icon: ['fal', 'save'],
        onClick: () => {
          console.log('Save clicked');
        },
      },
    ]}
  />
</Canvas>

### Different types of actions

<Canvas withSource='open'>
  <ActionBar
    title='title'
    actions={[
      {
        label: 'Submit',
        icon: ['fal', 'paper-plane'],
        color: ColorButtonEnum.PRIMARY,
        onClick: () => {
          console.log('Submit clicked');
        },
      },
    ]}
    basicActions={[
      {
        tooltip: 'Save',
        color: ColorButtonEnum.REVERSED,
        icon: ['fal', 'save'],
        onClick: () => {
          console.log('Save clicked');
        },
      },
    ]}
  />
</Canvas>
