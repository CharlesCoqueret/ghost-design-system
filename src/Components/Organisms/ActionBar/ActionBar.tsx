import React, { ReactElement } from 'react';
import classnames from 'classnames';

import { Button, ColorButtonEnum, IButtonProps } from '../../Molecules/';
import { Icon } from '../../Atoms';
import Title from './Title';

export interface IActionBarProps {
  /** Main actions (optional, default: undefined) */
  actions?: Array<IButtonProps>;
  /** Tooltip for the back button (optional, default: 'Back') */
  backTooltip?: string;
  /** Basic buttons displayed as icon (optional, default: undefined) */
  basicActions?: Array<IButtonProps>;
  /** Additional classname (optional, default: undefined) */
  className?: string;
  /** Identifier of the object being controlled (optional, default: undefined) */
  entityId?: string;
  /** Icon positioned after the title (optional, default: undefined) */
  icon?: ReactElement;
  /** Indicator positioned after the icon (optional, default: undefined) */
  indicator?: ReactElement;
  /**
   * Provides a way to go back. When defined, enables the back button (optional, default: undefined)
   * If using history (https://github.com/remix-run/history), it is highly recommended to implement history.go(-1) or history.back() behind this callback.
   */
  onBackClicked?: () => void;
  /** Handler of title change. When defined, enables edition of the title (optional, default: undefined) */
  onTitleEdit?: (newTitle: string) => void;
  /** Placeholder when the input value has been emptied (optional, default: undefined) */
  placeholder?: string;
  /** Must be true if the action bar if the primary bar */
  primary?: boolean;
  /** Prefix of the title, useful when a part of the title is not editable (optional, default: undefined) */
  prefix?: string;
  /** Rename tooltip indicating if the title is editable (optional, default: 'Rename') */
  renameTooltip?: string;
  /** Status positioned after the indidcator (optional, default: undefined) */
  status?: ReactElement;
  /** Suffix of the title, useful when a part of the title is not editable (optional, default: undefined) */
  suffix?: string;
  /** Action bar title */
  title: string;
}

const ActionBar = (props: IActionBarProps): ReactElement => {
  const {
    actions,
    backTooltip,
    basicActions,
    className,
    entityId,
    icon,
    indicator,
    onBackClicked,
    onTitleEdit,
    placeholder,
    primary,
    prefix,
    status,
    suffix,
    title,
  } = props;

  return (
    <div className={classnames('gds-action-bar', { primary: primary }, className)}>
      <div className='left-side'>
        {onBackClicked && (
          <div className='back-icon'>
            <Button
              icon={['fal', 'arrow-left']}
              onClick={onBackClicked}
              tooltip={backTooltip}
              color={ColorButtonEnum.REVERSED}
            />
          </div>
        )}
        <div className='title'>
          <Title
            entityId={entityId}
            onTitleEdit={onTitleEdit}
            placeholder={placeholder}
            prefix={prefix}
            suffix={suffix}
            title={title}
          />
        </div>
        {(icon || indicator || status) && (
          <div className='badges'>
            {icon}
            {indicator}
            {status}
          </div>
        )}
      </div>
      <div className='right-side'>
        {actions && actions.length > 0 && (
          <div className='actions'>
            {actions?.map((action) => {
              return <Button key={action.label || action.tooltip} {...action} icon={undefined} />;
            })}
          </div>
        )}
        {actions && actions.length > 0 && basicActions && basicActions.length > 0 && (
          <div className='separator'>
            <Icon icon={['fal', 'pipe']} size='lg' />
          </div>
        )}
        {basicActions && basicActions.length > 0 && (
          <div className='basic-actions'>
            {basicActions?.map((basicAction) => {
              return (
                <Button
                  key={basicAction.tooltip || basicAction.label}
                  {...basicAction}
                  color={ColorButtonEnum.REVERSED}
                />
              );
            })}
          </div>
        )}
      </div>
    </div>
  );
};

ActionBar.defaultProps = {
  actions: undefined,
  backTooltip: 'Back',
  basicActions: undefined,
  className: undefined,
  entityId: undefined,
  icon: undefined,
  indicator: undefined,
  onBackClicked: undefined,
  onTitleEdit: undefined,
  placeholder: undefined,
  primary: false,
  prefix: undefined,
  renameTooltip: 'Rename',
  status: undefined,
  suffix: undefined,
};

export default ActionBar;
