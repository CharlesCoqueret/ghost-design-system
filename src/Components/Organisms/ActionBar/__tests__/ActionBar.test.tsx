import React from 'react';
import { render } from '@testing-library/react';

import ActionBar from '../ActionBar';

describe('ActionBar Component', () => {
  it('ActionBar renders', async () => {
    const onTitleEditMock = jest.fn();

    const { container } = render(
      <ActionBar
        actions={[{ label: 'action1', dataTestId: 'ACTION1' }]}
        backTooltip='BACKTOOLTIP'
        basicActions={[{ label: 'basicaction1', dataTestId: 'BASICACTION1' }]}
        className='CLASSNAME'
        entityId='ENTITYID'
        icon={<div id='ICON' />}
        indicator={<div id='INDICATOR' />}
        onBackClicked={() => {
          return;
        }}
        onTitleEdit={onTitleEditMock}
        placeholder='PLACEHOLDED'
        primary={true}
        prefix='PREFIX'
        renameTooltip='TOOLTIP'
        status={<div id='STATUS' />}
        suffix='SUFFIX'
        title='TITLE'
      />,
    );

    expect(container).toMatchSnapshot();
    expect(onTitleEditMock).not.toBeCalled();
  });

  it('ActionBar renders without action, basicactions, icon, indicator or status', async () => {
    const onTitleEditMock = jest.fn();

    const { container } = render(
      <ActionBar
        backTooltip='BACKTOOLTIP'
        className='CLASSNAME'
        entityId='ENTITYID'
        onBackClicked={() => {
          return;
        }}
        onTitleEdit={onTitleEditMock}
        placeholder='PLACEHOLDED'
        primary={true}
        prefix='PREFIX'
        renameTooltip='TOOLTIP'
        suffix='SUFFIX'
        title='TITLE'
      />,
    );

    expect(container).toMatchSnapshot();
    expect(onTitleEditMock).not.toBeCalled();
  });
});
