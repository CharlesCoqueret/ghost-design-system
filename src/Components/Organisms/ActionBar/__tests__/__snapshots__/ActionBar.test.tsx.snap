// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Component ActionBar renders 1`] = `
<div>
  <div
    class="gds-action-bar primary CLASSNAME"
  >
    <div
      class="left-side"
    >
      <div
        class="back-icon"
      >
        <span>
          <button
            class="gds-button-content"
            color="reversed"
            type="button"
          >
            <div
              class="button-icon-container"
            >
              <i
                class="button-icon icon"
                icon="fal,arrow-left"
              />
            </div>
          </button>
        </span>
        <div>
          <div
            class="gds-tooltip bottom"
            style="left: -99999px; top: -99999px; opacity: 0;"
          >
            <div
              class="tooltip-arrow"
            />
            <div
              class="tooltip-label"
            >
              BACKTOOLTIP
            </div>
          </div>
        </div>
      </div>
      <div
        class="title"
      >
        <h1
          class="gds-typography ellipsis align-edit"
          title="PREFIX"
        >
          PREFIX
        </h1>
        <span
          style="display: inline-flex; margin: auto; overflow: hidden;"
        >
          <span
            class="title-edit-hidden"
          >
            TITLE
          </span>
          <input
            autocomplete="off"
            class="title-edit"
            placeholder="PLACEHOLDED"
            style="display: flex; width: 0px; margin: auto;"
            type="text"
            value="TITLE"
          />
        </span>
        <div>
          <div
            class="gds-tooltip bottom"
            style="left: -99999px; top: -99999px; opacity: 0;"
          >
            <div
              class="tooltip-arrow"
            />
            <div
              class="tooltip-label"
            >
              Rename
            </div>
          </div>
        </div>
        <h1
          class="gds-typography ellipsis align-edit"
        >
          - 
          ENTITYID
        </h1>
        <h1
          class="gds-typography ellipsis align-edit"
          title="SUFFIX"
        >
          SUFFIX
        </h1>
      </div>
      <div
        class="badges"
      >
        <div
          id="ICON"
        />
        <div
          id="INDICATOR"
        />
        <div
          id="STATUS"
        />
      </div>
    </div>
    <div
      class="right-side"
    >
      <div
        class="actions"
      >
        <button
          class="gds-button-content"
          color="secondary"
          data-testid="ACTION1"
          type="button"
        >
          <div
            class="button-label-container"
          >
            action1
          </div>
        </button>
      </div>
      <div
        class="separator"
      >
        <i
          class="icon"
          icon="fal,pipe"
        />
      </div>
      <div
        class="basic-actions"
      >
        <button
          class="gds-button-content"
          color="reversed"
          data-testid="BASICACTION1"
          type="button"
        >
          <div
            class="button-label-container"
          >
            basicaction1
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionBar Component ActionBar renders with actions without label, empty basicactions 1`] = `
<div>
  <div
    class="gds-action-bar"
  >
    <div
      class="left-side"
    >
      <div
        class="title"
      >
        <span
          class="title-edit-hidden"
        >
          TITLE
        </span>
        <input
          autocomplete="off"
          class="title-edit"
          readonly=""
          style="display: flex; width: 0px; margin: auto;"
          type="text"
          value="TITLE"
        />
      </div>
    </div>
    <div
      class="right-side"
    >
      <div
        class="actions"
      />
    </div>
  </div>
</div>
`;

exports[`ActionBar Component ActionBar renders without action, basicactions, icon, indicator or status 1`] = `
<div>
  <div
    class="gds-action-bar primary CLASSNAME"
  >
    <div
      class="left-side"
    >
      <div
        class="back-icon"
      >
        <span>
          <button
            class="gds-button-content"
            color="reversed"
            type="button"
          >
            <div
              class="button-icon-container"
            >
              <i
                class="button-icon icon"
                icon="fal,arrow-left"
              />
            </div>
          </button>
        </span>
        <div>
          <div
            class="gds-tooltip bottom"
            style="left: -99999px; top: -99999px; opacity: 0;"
          >
            <div
              class="tooltip-arrow"
            />
            <div
              class="tooltip-label"
            >
              BACKTOOLTIP
            </div>
          </div>
        </div>
      </div>
      <div
        class="title"
      >
        <h1
          class="gds-typography ellipsis align-edit"
          title="PREFIX"
        >
          PREFIX
        </h1>
        <span
          style="display: inline-flex; margin: auto; overflow: hidden;"
        >
          <span
            class="title-edit-hidden"
          >
            TITLE
          </span>
          <input
            autocomplete="off"
            class="title-edit"
            placeholder="PLACEHOLDED"
            style="display: flex; width: 0px; margin: auto;"
            type="text"
            value="TITLE"
          />
        </span>
        <div>
          <div
            class="gds-tooltip bottom"
            style="left: -99999px; top: -99999px; opacity: 0;"
          >
            <div
              class="tooltip-arrow"
            />
            <div
              class="tooltip-label"
            >
              Rename
            </div>
          </div>
        </div>
        <h1
          class="gds-typography ellipsis align-edit"
        >
          - 
          ENTITYID
        </h1>
        <h1
          class="gds-typography ellipsis align-edit"
          title="SUFFIX"
        >
          SUFFIX
        </h1>
      </div>
    </div>
    <div
      class="right-side"
    />
  </div>
</div>
`;
